Copy extended attributes if -p flag is provided to cp, mv. Contribute…

kjokinie opened the pull request:
…s to JB#30370

Doing a full file system copy of root is not currently possible
when busybox cp does not support copying extended attributes.

Added this patch from Dimitry Falko, which adds xattr support when using
"-p" option. This behaviour slightly differs from coreutils cp. Original cp
copying only ACLs and not copying executable file capabilities on "-p".
To copy all extended attributes with coreutils cp the option "--preserve=all"
must be used.

At this time of applying, the patch has not been merged to upstream busybox.

This patch was fixed by copying attributes after chown() called because it
resets file capabilities.

Signed-off-by: Igor Zhbanov igor.zhbanov@jolla.com

saukko commented on rpm/busybox-static.config:
```diff
@@ -1011,3 +1011,5 @@ CONFIG_FEATURE_LOGREAD_REDUCED_LOCKING=y
 CONFIG_FEATURE_KLOGD_KLOGCTL=y
 CONFIG_LOGGER=y
+
+CONFIG_XATTR=y
```
We should separate the config change to separate patch.

Commit:
Copy extended attributes if -p flag is provided to cp, mv. Contributes to JB#30370

Doing a full file system copy of root is not currently possible
when busybox cp does not support copying extended attributes.

Added this patch from Dimitry Falko, which adds xattr support when using
"-p" option. This behaviour slightly differs from coreutils cp. Original cp
copying only ACLs and not copying executable file capabilities on "-p".
To copy all extended attributes with coreutils cp the option "--preserve=all"
must be used.

At this time of applying, the patch has not been merged to upstream busybox.

This patch was fixed by copying attributes after chown() called because it
resets file capabilities.

Signed-off-by: Igor Zhbanov <igor.zhbanov@jolla.com>

Commit:
[rpm] Add XATTR as enabled by default. Contributes to JB#30370

Extended attributes are needed for cp to be able to copy rootfs
as it originally is. Enable CONFIG_XATTR.

Signed-off-by: Kalle Jokiniemi <kalle.jokiniemi@jolla.com>

saukko commented:
LGTM
